python/lsst/sims/maf/stackers/generalStackers.py

```diff
@@ -397,3 +399,45 @@ def _run(self, simData, cols_present=False):
         simData['season'] = season
         return simData
 
+
+class OpSimFieldStacker(BaseStacker):
+    """Add the fieldId of the closest OpSim field for each RA/Dec pointing.
+
+    Parameters
+    ----------
+    raCol : str, opt
+        Name of the RA column. Default fieldRA.
+    decCol : str, opt
+        Name of the Dec column. Default fieldDec.
+
+    """
+    def __init__(self, raCol='fieldRA', decCol='fieldDec', degrees=True):
+        self.colsReq = [raCol, decCol]
+        self.colsAdded = ['fieldId']
+        self.units = ['#']
+        self.raCol = raCol
+        self.decCol = decCol
+        self.degrees = degrees
+        fields_db = FieldsDatabase()
+        fieldid, ra, dec = fields_db.get_id_ra_dec_arrays("select * from Field;")  # Returned RA/Dec coordinates in degrees
```

Comment:
@rhiannonlynne and this is the utility function I use from that package to load fieldid, ra, dec.